### PR TITLE
Feature/3113/highlight nodes with area of zero

### PR DESCRIPTION
**Pull request by RomanenkoVladimir**
# Highlight files and folders with area metric of 0

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

closes: #3113 

## Description

 - Names and icons for folders and files with an area metric of zero/ unavailable area metric are written in gray
 - No context menu button is available for nodes with such an area metric value, a tooltip that explains that the area metric has no value for the node is provided instead
 - Right click context menu is also disabled in this case

## Screenshots or gifs

gray-text-for-unavailable-folder-metric

### CODE REVIEW

**Comment by RomanenkoVladimir** [comment]
The failing analyis-test seems flaky, as it fails across multiple PRs that introduce no changes to the analysis part of the application
Would be nice if someone could take a look to prevent it from blocking frontend PRs, or we could disable it

**Review by BridgeAR — COMMENTED** [review]
I really like how this looks!

File: visualization/app/codeCharta/ui/searchPanel/mapTreeView/mapTreeViewItemIcon/mapTreeViewItemIconColor.pipe.ts [hunk 1]
```diff
@@ -4,16 +4,23 @@ import { CodeMapNode } from "../../../../codeCharta.model"
...
 	static defaultColor = "#000000"
+	static areMetricZeroColor = "#BDBDBD"
 	static store = Store.store
 
 	transform(value: CodeMapNode): string | undefined {
-		if (isLeaf(value)) {
-			return undefined
+		const { areaMetric } = MapTreeViewItemIconColorPipe.store.getState().dynamicSettings
+
+		if (value.attributes[areaMetric]) {
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/searchPanel/mapTreeView/mapTreeViewItemIcon/mapTreeViewItemIconColor.pipe.ts:16]: Just a suggestion for more readability (less indentation due to early returns).

```suggestion
		if (!value.attributes[areaMetric]) {
			return MapTreeViewItemIconColorPipe.areMetricZeroColor
		}
		// ... other codnditions ...
```

File: visualization/app/codeCharta/ui/searchPanel/mapTreeView/mapTreeViewLevel/mapTreeViewLevel.component.html [hunk 2]
```diff
@@ -8,13 +8,14 @@
 		(mouseleave)="onMouseLeave()"
 		(click)="toggleOpen()"
 		(contextmenu)="openNodeContextMenu($event)"
+		[title]="node.attributes[areaMetric$ | async] > 0 ? '' : 'No Node Area for Chosen Metric'"
```
Comment (Hall-Ma) [inline visualization/app/codeCharta/ui/searchPanel/mapTreeView/mapTreeViewLevel/mapTreeViewLevel.component.html:11]:
```suggestion
		[title]="node.attributes[areaMetric$ | async] > 0 ? '' : 'No node area available for chosen metric'"
```

**Review by Hall-Ma — APPROVED** [review]
LGTM!

**Pull request merged**